### PR TITLE
feat: Integrate the context pane

### PR DESCRIPTION
**Pull request by dustinbyrne**
image

### CODE REVIEW

File: src/webviews/chatSearchWebview.ts [hunk 1]
```diff
@@ -137,6 +137,35 @@ export default class ChatSearchWebview {
           await vscode.commands.executeCommand('vscode.openWith', uri, 'appmap.views.appMapFile');
           break;
         }
+        case 'open-location': {
+          const { location } = message;
+          const [path, lineNumbers] = location.split(':');
```
Comment (ahtrotta) [inline src/webviews/chatSearchWebview.ts:142]: If this is an absolute path in Windows with a drive letter then this won't work correctly

Comment (dustinbyrne) [reply src/webviews/chatSearchWebview.ts]: Good catch! This is now fixed.

File: test/unit/util.test.ts [hunk 2]
```diff
@@ -32,3 +34,36 @@ describe('sanitizeEnvironment', () => {
...
+
+      expect(result instanceof URI).to.be.true;
+      expect(result.fsPath).to.equal(expected);
+    });
+  });
+
+  it('should parse a location with a line number', () => {
+    const examples = [
+      ['/path/to/file.rb:11', '/path/to/file.rb', 10],
+      ['C:\\path\\to\\file.rb:11', '\\path\\to\\file.rb', 10],
```
Comment (ahtrotta) [inline test/unit/util.test.ts:56]: Is it okay that the drive letter gets removed? If someone has multiple drives would VS Code be confused?

File: src/util.ts [hunk 3]
```diff
@@ -387,3 +387,19 @@ export function sanitizeEnvironment(env: NodeJS.ProcessEnv): Record<string, stri
   }
   return sanitizedEnv;
 }
+
+export function parseLocation(location: string): vscode.Location | vscode.Uri {
+  const match = location.match(/:(\d+(-\d+)?)$/);
+  if (match) {
+    const path = location.substring(0, match.index);
+    const [startLine, endLine] = match[1].split('-').map((x) => parseInt(x, 10) - 1);
```
Comment (ahtrotta) [inline src/util.ts:395]: The number after the colon for a data request is an event ID for an AppMap, so we need to handle that differently, right?

**Review by ahtrotta — APPROVED** [review]
Looks good and worked well when I tested it manually! :+1:

**Pull request merged**